multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/zeistmanga/ZeistManga.kt

```diff
@@ -149,31 +169,139 @@ abstract class ZeistManga(
         return GET(url.toString(), headers)
     }
 
-    override fun searchMangaSelector(): String = ".grid.gtc-f141a > div"
-    override fun searchMangaFromElement(element: Element): SManga {
-        return SManga.create().apply {
-            setUrlWithoutDomain(element.select(".block").attr("href"))
-            title = element.selectFirst(".clamp.toe.oh.block")!!.text().trim()
-            thumbnail_url = element.selectFirst("img")!!.attr("src")
-        }
-    }
+    override fun searchMangaParse(response: Response) = popularMangaParse(response)
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val url = "$baseUrl/search".toHttpUrl().newBuilder()
-            .addQueryParameter("q", query)
-            .build()
+        val startIndex = maxResults * (page - 1) + 1
+        val url = apiUrl()
+            .addQueryParameter("max-results", (maxResults + 1).toString())
+            .addQueryParameter("start-index", startIndex.toString())
 
-        return GET(url.toString(), headers)
-    }
+        if (query.isNotBlank()) {
+            url.addQueryParameter("q", query)
+                .build()
+
+            return GET(url.toString(), headers)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is StatusList -> {
+                    url.addPathSegment(filter.selected.value)
+                }
+
+                is TypeList -> {
+                    url.addPathSegment(filter.selected.value)
+                }
+
+                is LanguageList -> {
+                    url.addPathSegment(filter.selected.value)
+                }
+
+                is GenreList -> {
+                    filter.state.forEach { genre ->
+                        when (genre.state) {
+                            true -> url.addPathSegment(genre.value)
+                            false -> {}
+                        }
+                    }
+                }
+
+                else -> {}
+            }
+        }
 
-    override fun searchMangaNextPageSelector(): String? = null
+        return GET(url.build().toString(), headers)
+    }
 
     open fun apiUrl(feed: String = "Series"): HttpUrl.Builder {
         return "$baseUrl/feeds/posts/default/-/".toHttpUrl().newBuilder()
             .addPathSegment(feed)
             .addQueryParameter("alt", "json")
     }
 
+    override fun getFilterList(): FilterList {
+        if (!hasFilters) {
+            return FilterList(emptyList())
+        }
+        val filters = listOfNotNull(
+            Filter.Header(intl.filterWarning),
+            Filter.Separator(),
+            StatusList(intl.statusFilterTitle, getStatusList()),
+            TypeList(intl.typeFilterTitle, getTypeList()),
+            LanguageList(intl.languageFilterTitle, getLanguageList()),
+            GenreList(intl.genreFilterTitle, getGenreList()),
+        )
+        return FilterList(filters)
+    }
```

Comment:
You're not taking advantage of `listOfNotNull`, so you can rewrite like this.

```suggestion
    override fun getFilterList(): FilterList {
        if (!hasFilters) {
            return FilterList(emptyList())
        }
        
        return FilterList(
            Filter.Header(intl.filterWarning),
            Filter.Separator(),
            StatusList(intl.statusFilterTitle, getStatusList()),
            TypeList(intl.typeFilterTitle, getTypeList()),
            LanguageList(intl.languageFilterTitle, getLanguageList()),
            GenreList(intl.genreFilterTitle, getGenreList()),
        )
    }
```